Tools/run_batch.py

```
@@ -106,11 +122,120 @@ def _parse_run_config(path: Path) -> run_config_pb2.RunConfig:
+            communication_config.link_config,
+            f"Communication scenario {scenario.name} default link",
+        )
+
+        link_pairs = set()
+        for link_override in communication_config.link_overrides:
+            sender_type = getattr(link_override, "from")
+            receiver_type = link_override.to
+            if sender_type == 0 or receiver_type == 0:
+                raise ValueError(
+                    f"Communication scenario {scenario.name} contains an invalid link pair."
+                )
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)